### ISSUE TITLE
Embed human-readable channel name in buffer names?

### ISSUE THREAD

**Issue by ToxicFrog** [terminal-discord/weechat-discord#67]
The buffers are all internally named `weecord.<guild ID>.<channel ID>`. This means you can't `/buffer` to them except by index, since `/buffer` uses the full name, not the short name, for subname matching. For example, I have an `#off-topic` channel in one of my discords, and to get there, `/buffer off-topic` won't work; I have to use `/buffer .544728`.

It seems like it would be straightforward to fix this by embedding the short_name in the full buffer name, so it becomes `weecord.<guild>.<channel>.<name>`. Change `buffer_id_for_channel()` to take in the channel name as well as channel/guild IDs and slap it on the end. However, I'm not sure if this would interact badly with e.g. channel renames -- are there any obvious land mines here?

**Comment by Noskcaj19** [comment]
This is a bit of an interesting subject, the reason for the current format of `guild id.channel id` was due to the need to always be able to access a buffer for a given channel without having to do any network requests to find the names, avoid storing a separate index of buffers, and avoid iterating over all buffers.  This also never affected me much, as I use go.py and have it configured to use `short_name`.

However, I do not know if there are actually currently any cases in which the guild and channel names are not known that would make lookup difficult.  Additionally, there have been times when having a separate global store would be quite nice, and with recent changes to weechat-rs, that has been made a more reasonable option.

TLDR, I think the change would be great.  I don't believe weechat-discord has many internal landmines here, frankly I don't know that much about weechat and it's best practices, but I don't think there should by any there.
The one issue is that Discord allows for multiple guilds with the same name, as well as multiple channels _in the same guild_ with the same name.  This means we would probably have to stick some form of unique identifier in the id.

Frankly, weechats design of not separating internal and external details and concerns makes a lot of things rather difficult.

**Comment by ToxicFrog** [comment]
This issue actually prompted me to start using go.py and the UX there is so much nicer than /buffer that I no longer consider this a useful feature request, so feel free to close it. :)

**Comment by shanedabes** [comment]
I would definitely still find this useful. I use go.py, but I set my short names to emoji to fit loads on the bottom bar.
